src/modules/visits/controllers/visits.controller.ts

```diff
@@ -139,13 +148,35 @@ export class VisitsController {
 	}
 
 	@Patch(':id')
+	@ApiOperation({
+		description: 'Update a Visit request.',
+	})
+	@RequireAnyPermissions(Permission.UPDATE_VISIT_REQUEST, Permission.CANCEL_OWN_VISIT_REQUEST)
 	public async update(
 		@Param('id') id: string,
 		@Body() updateVisitDto: UpdateVisitDto,
-		@Session() session: Record<string, any>
+		@SessionUser() user: SessionUserEntity
 	): Promise<Visit> {
-		const user = SessionHelper.getArchiefUserInfo(session);
-		const visit = await this.visitsService.update(id, updateVisitDto, user.id);
+		if (
+			user.has(Permission.CANCEL_OWN_VISIT_REQUEST) &&
+			user.hasNot(Permission.UPDATE_VISIT_REQUEST)
+		) {
```

Comment:
isn't this flow possible? Seems simpler.
```
if (user.has(Permission.UPDATE_VISIT_REQUEST)) {
  // Update visit request
} else if (user.has(Permission.CANCEL_OWN_VISIT_REQUEST)) {
  // Check user is owner of visit request
  // Update visit request
}
```

Reply:
True, but this doubles the update logic...